src/main/java/com/efficio/fieldbook/web/study/germplasm/StudyEntryTransformer.java

```diff
@@ -50,151 +47,24 @@ public List<ImportedGermplasm> tranformToImportedGermplasm(final List<StudyEntry
 					importedGermplasm.setEntryTypeValue(entryType.get());
 					importedGermplasm.setEntryTypeCategoricalID(entryTypeCategoricalId);
 				}
-				importedGermplasm.setCross(studyEntryDto.getStudyEntryPropertyValue(TermId.CROSS.getId()).orElse(""));
+				importedGermplasm.setCross(studyEntryDto.getCross());
 				importedGermplasm.setDesig(studyEntryDto.getDesignation());
-				importedGermplasm.setEntryCode(studyEntryDto.getEntryCode());
+				importedGermplasm.setEntryCode(studyEntryDto.getStudyEntryPropertyValue(TermId.ENTRY_CODE.getId()).orElse(""));
 				importedGermplasm.setEntryNumber(studyEntryDto.getEntryNumber());
 				importedGermplasm.setGid(studyEntryDto.getGid().toString());
-				final Optional<String> groupGid = studyEntryDto.getStudyEntryPropertyValue(TermId.GROUPGID.getId());
+				final Optional<Integer> groupGid =  Optional.ofNullable(studyEntryDto.getGroupGid());
 				if (groupGid.isPresent()) {
 					final Integer mgid = Integer.valueOf(groupGid.get());
 					importedGermplasm.setMgid(mgid);
 					importedGermplasm.setGroupId(mgid);
 				}
-				importedGermplasm.setSource(studyEntryDto.getStudyEntryPropertyValue(TermId.SEED_SOURCE.getId()).orElse(""));
-				importedGermplasm.setGroupName(studyEntryDto.getStudyEntryPropertyValue(TermId.CROSS.getId()).orElse(""));
+				//FIXME: The GroupName is the Preferred Name of the groupGID
```

Comment:
@cuenyad anything we need to do with this FIXME in this branch? If not, please improve the comment to work on it later on

Reply:
done!.